src/Microsoft.DotNet.Darc/src/DarcLib/Models/Darc/DependencyFlowGraph.cs

```diff
@@ -127,20 +127,219 @@ public void RemoveEdge(DependencyFlowEdge edge)
             }
         }
 
-        public static DependencyFlowGraph Build(
+        /// <summary>
+        ///    Mark the back edges of the graph so that they can be ignored when walking it.
+        ///    Determine the backedges by computing dominators for each node. A node n is said to
+        ///    dominate another node if every path from the start to the other node must go through
+        ///    n.
+        /// </summary>
+        public void MarkBackEdges()
+        {
+            // In this, we will interpret the root of the graph as those nodes that have no outgoing edges
+            // (call them 'products').  Connect all of these via a start node.
+            // Add a start node to the graph (a node that connects all nodes that have no outgoing edges
+            // We will also reverse the graph by flipping the interpretation of incoming and outgoing.
+            DependencyFlowNode startNode = new DependencyFlowNode("start", "start");
+            foreach (var node in Nodes)
```

Comment:
>var [](start = 21, length = 3)

There is a lot of var's here that go against the coding style.  I think I can guess at... most of them? But we should be explicit.

Reply:
I'll clean it up.